### PR TITLE
Update beats

### PR DESCRIPTION
**Pull request by jalvz**
This update includes packaging refactor (https://github.com/elastic/beats/pull/7388)

required before https://github.com/elastic/apm-server/pull/1010

### CODE REVIEW

**Comment by jalvz** [comment]
@andrewkroh thanks for explicitly testing the new packaging with APM, seems to work smoothly 👏

**Comment by andrewkroh** [comment]
I'm happy to help with any changes. Here's a summary of what I think needs to be done.

1. Add magefile.go from this [gist](https://gist.github.com/andrewkroh/15d81ec0d04a4678fbf031af458cddf1).
1. Add package/README.md.tmpl from the gist. Where to put it is up to you. If you change the location then update the path in magefile.go.
1. Apply the patch to Makefile from the gist.
1. Update vendor/ to include:
    - github.com/elastic/beats/dev-tools/mage
    - github.com/magefile/mage/^
    - golang.org/x/tools/go/vcs

  This enables `make mage` to work and satisfies the dependencies of `magefile.go`.

## Testing

- [ ] Run `make mage` and test that `mage` is installed to `$GOPATH/bin/mage`.

- [ ] Test that `mage -l` works and shows a list of build targets.

- [ ] Test `make snapshot` and verify snapshots are in build/distributions.

- [ ] Test `make release` and verify non-snapshots are in build/distributions.

## External Changes

I would hold off on merging this in until after we have tested and verified everything with Beats.

release-manager
- https://github.com/elastic/release-manager/pull/364/files

apm-server-docker
- https://github.com/elastic/beats-docker/pull/40/files
- https://github.com/elastic/apm-server-docker/blob/master/Makefile#L85

jenkins
- https://github.com/elastic/infra/blob/master/ci/jjb/apm-ci/globals/templates/elastic%2Bapm-server/package.yml#L32
- https://github.com/elastic/infra/blob/master/ci/jjb/apm-ci/globals/macros/builder.yml#L219
- https://github.com/elastic/infra/pull/5765
- https://github.com/elastic/infra/pull/5766/files

**Comment by jalvz** [comment]
Many thanks @andrewkroh, done after some minor tweakings, tested the release targets as well and tried the darwin build.

https://github.com/elastic/apm-server-docker/pull/20
https://github.com/elastic/infra/pull/5798
https://github.com/elastic/release-manager/pull/365

Anything else you found out lately?

**Comment by jalvz** [comment]
@simitt @graphaelli would be great if any of you has a look as well when you have some time

File: Makefile [hunk 1]
```diff
@@ -11,12 +10,10 @@ TEST_ENVIRONMENT=true
 ES_BEATS?=./_beats
 PREFIX?=.
```
Comment (andrewkroh) [inline Makefile]: I think PREFIX is no longer used anywhere.

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,4 @@
 BEAT_NAME=apm-server
-BEAT_DESCRIPTION=Elastic Application Performance Monitoring Server
```
Comment (simitt) [inline Makefile]: From what I saw this would be actually used in `mage`.  Where does the description now come from?

**Pull request merged**